inductiva/mixins/file_manager.py

```diff
@@ -0,0 +1,94 @@
+"""Mixin for file management."""
+import os
+import shutil
+import pathlib
+from absl import logging
+
+from inductiva.utils import files, misc, templates
+
+
+class FileManager:
+    """Class for file management."""
+    __root_dir = None  #pylint: disable=invalid-name
+
+    def __check_root_dir(self):
+        if self.__root_dir is None:
+            raise ValueError("Root directory not set.")
+
+    def set_root_dir(self, root_dir=None):
+        """Set the root directory for the manager."""
+
+        if root_dir is None:
+            logging.info("Root directory not set. "
+                         "Setting default root_dir to be `input_dir`.")
+            root_dir = f"input_dir-{misc.create_random_tag()}"
+        elif os.path.isdir(root_dir):
+            logging.info(
+                "Directory %s already exists."
+                " Adding random tag to directory name.", root_dir)
+            root_dir = f"{root_dir}-{misc.create_random_tag(size=5)}"
```

Comment:
random tags are hard to read. People like to have a sense of sequence and this is why I suggest adding a sequential number.

This is the kind of logic we need:

```python
def gen_suffix(name, filenames):
    regex = f'^{name}#([0-9]+$)'

    max_suffix = -1
    for filename in filenames:
        if match := re.match(regex, filename):
            max_suffix = max(max_suffix, int(match.group(1)))

    exists_solo = name in filenames
    exists_derived = max_suffix >= 0

    if exists_derived:
        suffix = f'#{max_suffix+1}'
    elif exists_solo:
        suffix = '#2'
    else:
        suffix = ''
    return suffix

def gen_name(name):
    name = name.strip()
    filenames = os.glob.glob(f'{name}*')
    return name + gen_suffix(name, filenames)
```

```suggestion
        elif os.path.isdir(root_dir):
            new_root_dir = gen(root_dir)
            logging.info(
                "Directory %s already exists."
                " Setting root dir to %s", root_dir, new_root_dir)
            root_dir = new_root_dir"
```

(naming might no be the most adequate).

Reply:
BTW, the reason I extract the `gen_suffix` logic to a separate method is because we can easily test `gen_suffix` without having to work with the file system. It is easily testable, whereas `gen_name` requires real folders in the fs. This is a good pattern to keep in mind when testing is important.